pymatgen/io/lobster/outputs.py

```diff
@@ -934,188 +945,197 @@ def __init__(self, filename: str | None, **kwargs) -> None:
-    def _has_fatband(data):
-        for row in data:
-            splitrow = row.split()
-            if len(splitrow) > 1 and splitrow[1] == "FatBand":
+    def _has_fatband(data: list[str]) -> bool:
+        """Check whether calculation has hatband data."""
+        for line in data:
+            line_parts = line.split()
+            if len(line_parts) > 1 and line_parts[1] == "FatBand":
                 return True
         return False
 
     @staticmethod
-    def _get_dft_program(data):
-        for row in data:
-            splitrow = row.split()
-            if len(splitrow) > 4 and splitrow[3] == "program...":
-                return splitrow[4]
+    def _get_dft_program(data: list[str]) -> str | None:
+        """Get the DFT program used for calculation."""
+        for line in data:
+            line_parts = line.split()
+            if len(line_parts) > 4 and line_parts[3] == "program...":
+                return line_parts[4]
         return None
 
     @staticmethod
-    def _get_number_of_spins(data):
+    def _get_number_of_spins(data: list[str]) -> Literal[1, 2]:
+        """Get index of spin channel."""
         return 2 if "spillings for spin channel 2" in data else 1
 
     @staticmethod
-    def _get_threads(data):
-        for row in data:
-            splitrow = row.split()
-            if len(splitrow) > 11 and splitrow[11] in {"threads", "thread"}:
-                return splitrow[10]
+    def _get_threads(data: list[str]) -> int:
+        """Get number of CPU threads."""
+        for line in data:
+            line_parts = line.split()
+            if len(line_parts) > 11 and line_parts[11] in {"threads", "thread"}:
+                return int(line_parts[10])
         raise ValueError("Threads not found.")
 
     @staticmethod
-    def _get_spillings(data, number_of_spins):
-        charge_spilling = []
-        total_spilling = []
-        for row in data:
-            splitrow = row.split()
-            if len(splitrow) > 2 and splitrow[2] == "spilling:":
-                if splitrow[1] == "charge":
-                    charge_spilling += [np.float64(splitrow[3].replace("%", "")) / 100.0]
-                if splitrow[1] == "total":
-                    total_spilling += [np.float64(splitrow[3].replace("%", "")) / 100.0]
-
-            if len(charge_spilling) == number_of_spins and len(total_spilling) == number_of_spins:
+    def _get_spillings(
+        data: list[str],
+        number_of_spins: Literal[1, 2],
+    ) -> tuple[list[np.float64], list[np.float64]]:
+        """Get charge spillings and total spillings."""
+        charge_spillings = []
+        total_spillings = []
+        for line in data:
+            line_parts = line.split()
+            if len(line_parts) > 2 and line_parts[2] == "spilling:":
+                # TODO: DanielYang: why do we need float64?
+                if line_parts[1] == "charge":
+                    charge_spillings.append(np.float64(line_parts[3].replace("%", "")) / 100.0)
```

Comment:
I assume we don't need `float64` for spilling, which has only two decimal places:
```
spillings for spin channel 1
abs. total  spilling:   8.93%
abs. charge spilling:   0.98%
```